### PR TITLE
Adding windows tts (base implementation)

### PR DESCRIPTION
**Pull request by igubanov**
that PR adds a base implemetation of Windows TTS. that can be used optional, by default its unused

### CODE REVIEW

**Comment by Danily07** [comment]
Great work! <3 
There are just few comments mostly related with code style

**Comment by ivang7** [comment]
> Great work! <3 There are just few comments mostly related with code style

No promblem what things I can improve?

**Review by Danily07 — CHANGES_REQUESTED** [review]

**Comment by Danily07** [comment]
> > Great work! <3 There are just few comments mostly related with code style
> 
> No problem what things I can improve?

Sorry, forgot to submit comments :)

File: src/Translumo.TTS/Engines/WindowsTTSEngine.cs [hunk 1]
```diff
@@ -0,0 +1,28 @@
+﻿using System.Globalization;
+using System.Speech.Synthesis;
+
+namespace Translumo.TTS.Engines;
+
+public class WindowsTTSEngine : ITTSEngine
+{
+    private readonly string languageCode;
+    private SpeechSynthesizer synthesizer;
```
[the remark this answers is not in the document]
Comment (ivang7) [reply src/Translumo.TTS/Engines/WindowsTTSEngine.cs:9]: Ok, i dont like _ prefix for the private field, but I have seen that this project required that, I will fix it

File: src/Translumo.Processing/TranslationProcessingService.cs [hunk 2]
```diff
@@ -376,6 +386,15 @@ private void TranslationConfigurationOnPropertyChanged(object sender, PropertyCh
             _translator = _translatorFactory.CreateTranslator(_translationConfiguration);
         }
 
+        private void TtsConfigurationOnPropertyChanged(object sender, PropertyChangedEventArgs e)
+        {
+            if (e.PropertyName == nameof(_ttsConfiguration.TtsLanguage)
+                || e.PropertyName == nameof(_ttsConfiguration.TtsSystem))
+            {
+                _ttsEngine = _ttsFactory.CreateTtsEngine(_ttsConfiguration);
```
[the remark this answers is not in the document]
Comment (ivang7) [reply src/Translumo.Processing/TranslationProcessingService.cs:394]: Ok I will check it and add correct destroy

File: src/Translumo.TTS/Engines/NoneTTSEngine.cs [hunk 3]
```diff
@@ -0,0 +1,12 @@
+﻿namespace Translumo.TTS.Engines;
+
+public class NoneTTSEngine : ITTSEngine
```
[the remark this answers is not in the document]
Comment (ivang7) [reply src/Translumo.TTS/Engines/NoneTTSEngine.cs:3]: A null value is a dirty solution, I think that empty/null object is more flexible and oop approach, let's try use the object for "disabling" tts

File: src/Translumo.Infrastructure/Powershell/OptionalFeaturesProvider.cs [hunk 4]
```diff
@@ -37,18 +57,24 @@ public static async Task<LanguagePackInstallResult> OcrLanguagePackInstall(strin
...
+                    RestartIsNeeded = (bool)languagePackPsObject.Properties["RestartNeeded"].Value
                 };
             }
         }
 
-        private static async Task<PSObject> GetPsObjectOcrLanguagePack(string languageCode)
+        private static async Task<PSObject> GetPsObjectOcrLanguagePack(string languageCode) =>
+            await GetPsObjectLanguagePack(languageCode, "OCR");
+
+        private static async Task<PSObject> GetPsObjectTextToSpeechLanguagePack(string languageCode) =>
```
[the remark this answers is not in the document]
Comment (ivang7) [reply src/Translumo.Infrastructure/Powershell/OptionalFeaturesProvider.cs:68]: Sorry, I don't understand, I dont want to add more parameters in the public signature, client should not know our internal logic, and in those cases I think one more method is better than one more parameter

[the remark this answers is not in the document]
Comment (Danily07) [reply src/Translumo.Infrastructure/Powershell/OptionalFeaturesProvider.cs:68]: No need to add more parameters in the public signature, I mean in public TtsLanguagePackIsInstalled method just directly call GetPsObjectLanguagePack(languageCode, "TextToSpeech"), but replace string literal with a private enum

File: src/Translumo.TTS/Engines/NoneTTSEngine.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (Danily07) [reply src/Translumo.TTS/Engines/NoneTTSEngine.cs:3]: Well, an object, by definition, that will never contain an implementation is some sort of code smell, but I agree that without nullable reference type my suggestion is not the best. Let's leave current implementation until I refactor solution with enabled nullable reference type

File: src/Translumo.TTS/Engines/WindowsTTSEngine.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (igubanov) [reply src/Translumo.TTS/Engines/WindowsTTSEngine.cs]: fix and add rules to .editorconfig

File: src/Translumo.Infrastructure/Powershell/OptionalFeaturesProvider.cs (hunk 4 above)
[the remark this answers is not in the document]
Comment (Danily07) [reply src/Translumo.Infrastructure/Powershell/OptionalFeaturesProvider.cs:68]: Okay, I just checked GetPsObjectLanguagePack implementation once again. Current imlementation is fine, I will close the comment :)

File: src/Translumo.Processing/TranslationProcessingService.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (igubanov) [reply src/Translumo.Processing/TranslationProcessingService.cs:395]: Yes SpeechSynthesizer  is disposable, I add IDisposable to ITTSEngine and dispose it in the TranslationProcessingService, so it also IDisposable

May be it is over engineering, but I usually use IDisposable object not far than in the same method/scope

**Pull request merged**